### ISSUE TITLE
Questions about subgoal evaluation of validation set. 

### ISSUE THREAD

**Issue by xxxfrank** [askforalfred/alfred#27]
Hi, First thanks your awesome work. 
I have encountered some problems when i evaluated  the following pre-trained model.
![image](https://user-images.githubusercontent.com/34808562/83108601-9fb21780-a0f2-11ea-8160-18b70633f71d.png)
### environment
![image](https://user-images.githubusercontent.com/34808562/83108797-e4d64980-a0f2-11ea-9574-2e4ffd78d7eb.png)

I have evaluated the model with following commands.

```
python3 models/eval/eval_seq2seq.py --model_path ./data/model\:seq2seq_im_mask\,name\:base30_pm010_sg010_01/best_seen.pth --eval_split valid_seen --data ./data/json_feat_2.1.0/ --model models.model.seq2seq_im_mask --gpu --num_threads 1 --subgoals all
```

Expected  output:
![image](https://user-images.githubusercontent.com/34808562/83108876-059e9f00-a0f3-11ea-9538-617a14ea90b8.png)

My output:
```
Avg = 40.58
{'CleanObject': {'evals': 112,                                                                                                                                           
                 'sr': 0.17857142857142858,
                 'sr_plw': 0.17857142857142858,
                 'successes': 20},
 'CoolObject': {'evals': 132,
                'sr': 0.8333333333333334,
                'sr_plw': 0.8160280299410734,
                'successes': 110},
 'GotoLocation': {'evals': 2660,
                  'sr': 0.45338345864661656,
                  'sr_plw': 0.3228794854619464,
                  'successes': 1206},
 'HeatObject': {'evals': 107,
                'sr': 0.822429906542056,
                'sr_plw': 0.7851184079299491,
                'successes': 88},
 'PickupObject': {'evals': 1211,
                  'sr': 0.19240297274979357,
                  'sr_plw': 0.15914573049097747,
                  'successes': 233},
 'PutObject': {'evals': 1103,
               'sr': 0.642792384406165,
               'sr_plw': 0.5785606947178042,
               'successes': 709},
 'SliceObject': {'evals': 151,
                 'sr': 0.33112582781456956,
                 'sr_plw': 0.24598930481283418,
                 'successes': 50},
 'ToggleObject': {'evals': 94,
                  'sr': 0.18085106382978725,
                  'sr_plw': 0.16046099290780141,
                  'successes': 17}}
```

I am looking forward to your reply.

**Comment by MohitShridhar** [comment]
@xxxfrank are you using the right version of THOR from [requirements.txt](https://github.com/askforalfred/alfred/blob/master/requirements.txt). 

Also, one user reported that he had to [train the model](https://github.com/askforalfred/alfred/tree/master/models#training) himself, instead of using the pre-trained model to match the numbers. Can you try this, while I look into it?

**Comment by xxxfrank** [comment]
Thanks for your reply. I will train the model myself.

The THOR-version is 2.1.0 and requirement package was installed with  
```
pip3 install -r requirments.txt
```
I wonder whether the pre-trained res18 is the right version, which has been downloaded  from url during test?
```
https://download.pytorch.org/models/resnet18-5c106cde.pth
```

**Comment by MohitShridhar** [comment]
I am also using `resnet18-5c106cde.pth`. I wonder if something else changed in pytorch or torchvision.

**Comment by xxxfrank** [comment]
@MohitShridhar Hi, i have trained the model myself and evaluate it on valid-seen dataset.
The difference between this version and expected output is marginal. 
```
CleanObject ==========
SR: 95/112 = 0.848
PLW SR: 0.816
CoolObject ==========
SR: 120/132 = 0.909
PLW SR: 0.898
GotoLocation ==========
SR: 1701/2660 = 0.639
PLW SR: 0.530
HeatObject ==========
SR: 94/107 = 0.879
PLW SR: 0.840
PickupObject ==========
SR: 559/1211 = 0.462
PLW SR: 0.396
PutObject ==========
SR: 934/1103 = 0.847
PLW SR: 0.833
SliceObject ==========
SR: 47/151 = 0.311
PLW SR: 0.235
ToggleObject ==========
SR: 94/94 = 1.000
PLW SR: 1.000
```

**Comment by MohitShridhar** [comment]
Interesting, thanks for checking this out. I'll see if updating it with a retrained model from a fresh clone helps.

**Comment by k15201363625** [comment]
> @MohitShridhar Hi, i have trained the model myself and evaluate it on valid-seen dataset. The difference between this version and expected output is marginal.
> 
> ```
> CleanObject ==========
> SR: 95/112 = 0.848
> PLW SR: 0.816
> CoolObject ==========
> SR: 120/132 = 0.909
> PLW SR: 0.898
> GotoLocation ==========
> SR: 1701/2660 = 0.639
> PLW SR: 0.530
> HeatObject ==========
> SR: 94/107 = 0.879
> PLW SR: 0.840
> PickupObject ==========
> SR: 559/1211 = 0.462
> PLW SR: 0.396
> PutObject ==========
> SR: 934/1103 = 0.847
> PLW SR: 0.833
> SliceObject ==========
> SR: 47/151 = 0.311
> PLW SR: 0.235
> ToggleObject ==========
> SR: 94/94 = 1.000
> PLW SR: 1.000
> ```
Did you change the training parameters? Why are the results so much better even compared to those in the paper?
